pegasus/sites.v3/code.org/public/js/extended_us_map.js

```diff
@@ -794,16 +794,16 @@ document.addEventListener("DOMContentLoaded", function() {
   console.log("DOMContentLoaded");
 });
 
-$(document).ready(function () {
-  initExtendedMap();
-});
-
 // NOTE: This is not a good pattern to replicate, but is a workaround
 // for an issue with loading Raphael before attempting to render the map.
 // A better implementation would move this code to /apps/src.
 
 let initAttemptsRemaining = 10;
 
+$(document).ready(function () {
+  initExtendedMap();
+});
```

Comment:
I have to say that I don't really understand why in the 4th attempt, initExtendedMap would be defined but initAttemptsRemaining wouldn't be. Unless you know otherwise, I would suggest moving the `$(document).ready` call to below the definition of renderMap, since that seems more likely to make sure that initExtendedMap and renderMap are also defined.

Reply:
Ah, it looks like it's the way [hoisting](https://www.sitepoint.com/demystifying-javascript-variable-scope-hoisting/) works: the variable declaration is hoisted to global scope, but it can't be read/written until its initial value has been assigned.

I was able to confirm with a locally simulated repro:

```
initExtendedMap();

let initAttemptsRemaining = 10;

function initExtendedMap() {
  if (initAttemptsRemaining === 0) {
    return;
  }
```

And verified that the change in this PR fixes it.

Reply:
Oh, and from the same article, it appears that function declarations are completely hoisted, which is why they are in scope here.

Reply:
I'm sold!